framework/src/Volo.Abp.Cli.Core/Volo/Abp/Cli/ProjectModification/AngularPwaSupportAdder.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using Volo.Abp.Cli.Utils;
+using Volo.Abp.DependencyInjection;
+
+namespace Volo.Abp.Cli.ProjectModification;
+
+public class AngularPwaSupportAdder : ITransientDependency
+{
+    protected ICmdHelper CmdHelper { get; }
+    protected PackageJsonFileFinder PackageJsonFileFinder { get; }
+
+    public AngularPwaSupportAdder(
+        ICmdHelper cmdHelper,
+        PackageJsonFileFinder packageJsonFileFinder)
+    {
+        CmdHelper = cmdHelper;
+        PackageJsonFileFinder = packageJsonFileFinder;
+    }
+
+    public virtual void AddPwaSupport(string rootDirectory)
+    {
+        var fileList = PackageJsonFileFinder.Find(rootDirectory).Where(x => x.Contains("angular.json")).ToList();
```

Comment:
This won't work. `x` is the full path of `appsettings.json` file, you shouldn't filter it with `angular.json`.

```suggestion
       PackageJsonFileFinder.Find(rootDirectory).Where(x => File.Exist( x.RemovePostFix("package.json") + "angular.json") ).ToList();
```